### PR TITLE
Adding bdk keymap for ergobox_ez

### PR DESCRIPTION
**Pull request by bendk**
## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by this PR

* 

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

I believe my code follows the code style, but please correct me if I'm wrong.

### CODE REVIEW

File: keyboards/ergodox_ez/keymaps/bdk/config.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/bdk/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/ergodox_ez/keymaps/bdk/config.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+#define CONFIG_USER_H
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/bdk/config.h]:
```suggestion
```

File: keyboards/ergodox_ez/keymaps/bdk/config.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+#include "../../config.h"
+
+#define ONESHOT_TAP_TOGGLE 2
+#define ONESHOT_TIMEOUT 167
+#define FORCE_NKRO
+
+#endif
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/bdk/config.h]:
```suggestion
```

File: keyboards/ergodox_ez/keymaps/bdk/config.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#ifndef CONFIG_USER_H
+#define CONFIG_USER_H
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/bdk/config.h]: This line is not actually needed, and should be removed. 

```suggestion
```

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/ergodox_ez/keymaps/bdk/keymap.c [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+// Layers
+enum {
+        BASE=0,
+        SYMNUM,  // Symbols and numbers (activated by holding enter)
+        FKEYS,   // Function keys (activated by a thumb button)
+};
+
+// counter that increments on all key press events.  This is needed for our LT
+// handling
+static uint16_t press_counter = 0;
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/bdk/keymap.c]: This doesn't actually need to be static.

File: keyboards/ergodox_ez/keymaps/bdk/keymap.c [hunk 6]
```diff
@@ -0,0 +1,176 @@
+#include QMK_KEYBOARD_H
+
+#define SHIFT_DOWN() (get_mods() & (MOD_BIT(KC_LSFT)|MOD_BIT(KC_RSFT)))
+#define KC_FKEYS MO(FKEYS)
+
+// Layers
+enum {
+        BASE=0,
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/bdk/keymap.c]: This is oddly indented.

```suggestion
  BASE=0,
```

**Comment by bendk** [comment]
Thanks for the quick review!  I believe I fixed all the issues you pointed out.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**